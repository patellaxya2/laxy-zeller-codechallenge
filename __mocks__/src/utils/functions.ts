
const validateName = jest.fn();
const validateEmail = jest.fn();
export { validateEmail, validateName };
