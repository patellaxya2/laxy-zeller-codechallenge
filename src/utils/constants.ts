

export const roles = ["All","Admin", "Manager"]



