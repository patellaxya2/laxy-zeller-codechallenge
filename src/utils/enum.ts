export enum EROLES{
    ALL="ALL",
    ADMIN = "ADMIN",
    MANAGER="MANAGER"
    
}
