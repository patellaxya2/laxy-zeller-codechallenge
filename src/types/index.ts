import { EROLES } from "../utils/enum";

 export interface IZellerCustomer  {
    id: string;
    name: string;
   role: EROLES;
   email: string;
};
  


export type StackParamList= {
  Dashboard: undefined;
  AddNewUser: undefined;
};


export interface IGetZellerCustomersResponse {
  listZellerCustomers: {
    items: IZellerCustomer[];
  };
}