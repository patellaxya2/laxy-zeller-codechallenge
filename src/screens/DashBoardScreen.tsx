import React, { useRef, useState } from 'react'
import { StatusBar, StyleSheet, Text, View, Animated, SectionList, TouchableOpacity, useAnimatedValue, Button, Image, TextInput } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import theme from '../styles/theme'
import { NativeStackScreenProps } from '@react-navigation/native-stack'
import { StackParamList } from '../types'
import PagerView from 'react-native-pager-view';
import { roles, } from '../utils/constants';
import { EROLES } from '../utils/enum';
import { IZellerCustomer } from '../types';
import UserItem from '../components/UserItem';
import AnimatedTabView from '../components/AnimatedTabView'
import useTabAnimation from '../hooks/useTabAnimation'
import { useQuery } from '../realm/setup'
import { ZellerCustomer } from '../realm/models/ZellerCustomer'
import { Results } from 'realm'


type DashBoardScreenProps = NativeStackScreenProps<StackParamList, 'Dashboard'>;

const DashBoardScreen: React.FC<DashBoardScreenProps> = ({ navigation }) => {
    const zellerCustomerList = useQuery(ZellerCustomer);
    const { translateX, startAnimation, tabWidth } = useTabAnimation({ items: roles });
    const [selected, setSelected] = useState<string>(roles[0]);
    const pagerRef = useRef<PagerView>(null);
    const [searchName, setSearchName] = useState("")

    const animateToTab = (index: number) => {
        startAnimation(index, 300);
    };

    const handleTabPress = (role: string, index: number) => {
        setSelected(role);
        pagerRef.current?.setPage(index);
        animateToTab(index);
    };

    const handlePageSelected = (e: any) => {
        const index = e.nativeEvent.position;
        const role = roles[index];
        setSelected(role);
        animateToTab(index);

    };
    const groupByLetter = (data: IZellerCustomer[]) => {
        const sorted = [...data].sort((a, b) => a.name.localeCompare(b.name));
        const grouped: { title: string; data: IZellerCustomer[] }[] = [];

        sorted.forEach((item) => {
            const letter = item.name[0].toUpperCase();
            const section = grouped.find((g) => g.title === letter);
            if (section) {
                section.data.push(item);
            } else {
                grouped.push({ title: letter, data: [item] });
            }
        });

        return grouped;
    };
    const renderList = (filter: EROLES, mZellerCustomerList: Results<ZellerCustomer>, searchByName: string) => {


        let filtered = mZellerCustomerList;
        if (searchByName && filter && filter === "ALL") {
            filtered = mZellerCustomerList.filtered('name CONTAINS[c] $0'
                , searchByName)
        }
        else if (searchByName && filter && filter !== "ALL") {
            filtered = mZellerCustomerList.filtered('role == $0 AND name CONTAINS[c] $1'
                , filter, searchByName)
        }
        else if (filter !== "ALL") {
            filtered = mZellerCustomerList.filtered(`role == $0`, filter)
        }

        const filteredUI: IZellerCustomer[] = Array.from(filtered).map((item) => ({
            id: item?.id,
            name: item.name,
            role: item.role,
            email: item.email,
        }));
        const sections = groupByLetter(filteredUI);

        return (
            <SectionList

                testID='SectionList'
                sections={sections}
                keyExtractor={(item) => item.id}
                renderItem={({ item }) => <UserItem item={item} />}
                renderSectionHeader={({ section: { title } }) => (
                    <Text style={styles.sectionHeader}>{title}</Text>
                )}
                contentContainerStyle={{ paddingBottom: 80 }}
                ListEmptyComponent={() => <Text style={{ textAlign: "center" }}>Data Not Found!</Text>}

            />
        );
    };



    return (
        <SafeAreaView style={styles.container}>
            <StatusBar
                barStyle={'dark-content'}
                backgroundColor={theme.colors.primary}
            />


            <AnimatedTabView
                items={roles}
                handleTabPress={handleTabPress}
                selected={selected}
                tabWidth={tabWidth}
                translateX={translateX as Animated.Value}
            />


            <TextInput
                placeholder='Search by name...'
                style={{
                    alignSelf: "center",
                    width: '85%',
                    borderWidth: 1,
                    borderColor: theme.colors.light_gray,
                    padding: 10,
                    marginVertical: 15,
                    borderRadius: 40

                }}
                onChangeText={(text) => {
                    setSearchName(text);
                }}
            />
            <PagerView
                testID='pager-view'
                style={styles.container}
                initialPage={0}
                onPageSelected={handlePageSelected}
                ref={pagerRef}
            >
                <View key="1">
                    {selected === roles[0] && renderList(EROLES.ALL, zellerCustomerList, searchName)}
                </View>
                <View key="2">
                    {selected === roles[1] && renderList(EROLES.ADMIN, zellerCustomerList, searchName)}
                </View>
                <View key="3">
                    {selected === roles[2] && renderList(EROLES.MANAGER, zellerCustomerList, searchName)}
                </View>
            </PagerView>
            <TouchableOpacity style={styles.fab} onPress={() => {
                navigation.navigate("AddNewUser")
            }}>
                <Text style={styles.fabText}>+</Text>
            </TouchableOpacity>



        </SafeAreaView>
    )
}

export default DashBoardScreen
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: theme.colors.primary
    },
    tabContainer: {
        alignItems: 'center',
        margin: 10,
    },
    tabs: {
        flexDirection: "row",
        justifyContent: "center",
        height: 40,
        backgroundColor: theme.colors.smoke_gray,
        borderRadius: 20,
        width: "80%",
        position: 'relative',
        // overflow: 'hidden'
    },
    tab: {
        flex: 1,
        // paddingHorizontal: 16,
        justifyContent: "center",
        // marginHorizontal: 4,
        // backgroundColor: "orange",
        alignItems: "center",
        // zIndex: 2, // Ensure tabs are above the indicator
    },
    tabText: {
        fontSize: 14,
        color: "#333",
    },
    tabTextActive: {
        color: theme.colors.secondary,
        fontWeight: "600",
    },
    indicator: {
        position: 'absolute',
        height: "100%",
        borderColor: theme.colors.secondary, borderRadius: 20, borderWidth: 1,
        left: 0,
        zIndex: 1,
    },
    tabActive: { borderColor: theme.colors.secondary, borderRadius: 20, borderWidth: 1, },

    sectionHeader: {
        paddingHorizontal: 16,
        paddingVertical: 6,
        fontWeight: "bold",
        backgroundColor: theme.colors.very_light_gray,
    },
    fab: {
        position: "absolute",
        bottom: 24,
        right: 24,
        width: 52,
        height: 52,
        borderRadius: 26,
        backgroundColor: theme.colors.secondary,
        justifyContent: "center",
        alignItems: "center",
        elevation: 3,
    },
    fabText: { fontSize: 28, color: theme.colors.primary, textAlign: "center" },
});