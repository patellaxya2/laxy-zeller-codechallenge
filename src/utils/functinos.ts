import Realm, { BSON } from "realm";
import { ZellerCustomer } from "../realm/models/ZellerCustomer";
import { IZellerCustomer } from "../types";

/**
 * Syncs backend data into Realm
 * @param realm The Realm instance
 * @param apiData The array of ZellerCustomer objects from backend
 */
export const syncZellerCustomers = async (
  realm: Realm,
  apiData: IZellerCustomer[]
) => {
  if (!apiData || apiData.length === 0) return;

  realm.write(() => {
    apiData.forEach((item) => {
      realm.create(
        ZellerCustomer,
        {
          id: item.id ,
          name: item.name,
          email: item.email,
          role: item.role, 
        },
        Realm.UpdateMode.Modified 
      );
    });
  });
};

export const validateName = (name: string) => {
  if (!name.trim()) return 'Name cannot be empty';  
  if (!/^[A-Za-z ]+$/.test(name)) return 'Only alphabets and spaces allowed';
  if (name.length > 50) return 'Name cannot exceed 50 characters';
  return '';
};

export const validateEmail = (email: string) => {
  if (!email) return '';
  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  if (!emailRegex.test(email)) return 'Invalid email format';
  return '';
};


export const groupByLetter = (data: IZellerCustomer[]) => {
  const sorted = [...data].sort((a, b) => a.name.localeCompare(b.name));
  const grouped: { title: string; data: IZellerCustomer[] }[] = [];

  sorted.forEach((item) => {
      const letter = item.name[0].toUpperCase();
      const section = grouped.find((g) => g.title === letter);
      if (section) {
          section.data.push(item);
      } else {
          grouped.push({ title: letter, data: [item] });
      }
  });

  return grouped;
};