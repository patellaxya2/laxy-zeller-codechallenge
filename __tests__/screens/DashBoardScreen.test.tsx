// __tests__/screens/DashBoardScreen.test.tsx
import React from "react";
import { render, fireEvent, within } from "@testing-library/react-native";
import DashBoardScreen from "../../src/screens/DashBoardScreen";
import { roles } from "../../src/utils/constants";
import { NativeStackNavigationProp, NativeStackScreenProps } from "@react-navigation/native-stack";
import { IZellerCustomer, StackParamList } from "../../src/types";
import { EROLES } from "../../src/utils/enum";
import { Text } from "react-native";


// Mock navigation
const mockNavigate = jest.fn();
const mockNavigation: Partial<NativeStackNavigationProp<StackParamList, 'Dashboard'>> = {
  navigate: mockNavigate,
};

// Mock route
const mockRoute: NativeStackScreenProps<StackParamList, 'Dashboard'>['route'] = {
  key: 'DashboardKey',
  name: 'Dashboard',
  params: undefined,
};

// Mock Realm hooks
jest.mock('../../src/realm/setup', () => ({
  RealmProvider: ({ children }: any) => children,
  useQuery: jest.fn(() => {
    const data = [
      { id: '1', name: 'Alice', role: 'ADMIN', email: 'alice@test.com' },
      { id: '2', name: 'Bob', role: 'MANAGER', email: 'bob@test.com' },
      { id: '3', name: 'Charlie', role: 'ADMIN', email: 'charlie@test.com' },
    ];
    return {
      filtered: (query: string, ...args: any[]) => {
        console.log('query: args  ', query, args);

        // const roleName = role.replace(/"/g, '').split('=')[1]?.trim() || role;

        // return role === 'ALL' ? true : item.role == roleName
        let result = data;
        if (query.includes(`role == $0`)) {
          result = result.filter((item) => item.role === args[0])
        }
        else if (query.includes('name CONTAINS[c] $0')) {
          const search = args[0].toLowerCase();
          result = result.filter(item => item.name.toLowerCase().includes(search));
        }
        else if (query.includes('role == $0 AND name CONTAINS[c] $1')) {
          const roleArg = args[0];
          const search = args[0].toLowerCase();
          result = result.filter(item => item.role === roleArg && item.name.toLowerCase().includes(search));
        }
        return result;
      }
      ,
      [Symbol.iterator]: function* () {
        for (const item of data) yield item;
      },
    }
  }),
  useRealm: jest.fn(),
  useObject: jest.fn(),
  useProgress: jest.fn(),
}))

// Mock useTabAnimation hook
const mockStartAnimation = jest.fn();
jest.mock('../../src/hooks/useTabAnimation', () => ({
  __esModule: true,
  default: jest.fn(() => ({
    translateX: 3,
    startAnimation: mockStartAnimation,
    tabWidth: 90,
  })),
}));

// Mock UserItem component
jest.doMock('../../src/components/UserItem', () => ({
  __esModule: true,
  default: ({ item }: { item: IZellerCustomer }) => <Text>{item.name}</Text>,
}));

// ------------------ TESTS ------------------ //

describe("DashBoardScreen Integration Tests", () => {
  let getByText: any;
  let queryByText: any;
  let getByTestId: any;

  beforeEach(() => {
    const rendered = render(<DashBoardScreen navigation={mockNavigation as any} route={mockRoute} />);
    getByText = rendered.getByText;
    queryByText = rendered.queryByText;
    getByTestId = rendered.getByTestId;
    jest.clearAllMocks();
  });

  it("renders all tabs", () => {
    roles.forEach((role) => {
      expect(getByTestId(`tab-${role}`)).toBeTruthy();
    });

    // Default ALL customers visible
    expect(getByText("Alice")).toBeTruthy();
    expect(getByText("Bob")).toBeTruthy();
    expect(getByText("Charlie")).toBeTruthy();
  });

  it("handles tab press animation", () => {
    const allTab = getByTestId('tab-Admin');
    fireEvent.press(allTab);

    expect(mockStartAnimation).toHaveBeenCalledWith(1, 300);
  });

  it("renders only ADMIN customers when filtered", () => {


    // Simulate tab press for ADMIN
    const adminTab = getByTestId('tab-Admin');
    fireEvent.press(adminTab);
    //    console.log('adminTab : ', adminTab);

    expect(getByText("Alice")).toBeTruthy();
    expect(getByText("Charlie")).toBeTruthy();;
    expect(queryByText("Bob")).toBeNull();;
  });



  it("navigates to AddNewUser when FAB is pressed", () => {
    const fab = getByText("+");
    fireEvent.press(fab);

    expect(mockNavigate).toHaveBeenCalledWith("AddNewUser");
  });
});

describe("handlePageSelected for pagerview", () => {


  it("should update selected role and call animateToTab when page is selected", () => {
    const { getByText, getByTestId } = render(<DashBoardScreen navigation={mockNavigate as any} route={mockRoute} />);

    // PagerView is rendered with a testID (add testID="pager-view" in your component if missing)
    const pager = getByTestId("pager-view");

    // Fire event to simulate selecting second page (index 1 â†’ roles[1])
    fireEvent(pager, "onPageSelected", {
      nativeEvent: { position: 1 },
    });

    const tab = getByTestId("tab-Admin");

    // Find the nested <Text> inside this TouchableOpacity
    const text = within(tab).getByText("Admin");
    // Expect role is updated in UI
    expect(text).toHaveStyle({
      color: "#007aff",
      fontWeight: "600",
    });

  });
});

describe('search functionality', () => {
  beforeEach(() => {

    jest.clearAllMocks();
  });
  it('filters customer list based on search input', () => {
    const { getByPlaceholderText, queryByText, getByText } = render(
      <DashBoardScreen navigation={mockNavigation as any} route={mockRoute} />
    );

    const searchInput = getByPlaceholderText('Search by name...');

    // Initially all users visible
    expect(getByText('Alice')).toBeTruthy();
    expect(getByText('Bob')).toBeTruthy();
    expect(getByText('Charlie')).toBeTruthy();

    // Type "Alice" in search
    fireEvent.changeText(searchInput, 'Alice');

    // Only Alice should be visible
    expect(getByText('Alice')).toBeTruthy();
    expect(queryByText('Bob')).toBeNull();
    expect(queryByText('Charlie')).toBeNull();

    // Type a non-existing name
    fireEvent.changeText(searchInput, 'NonExisting');

    // No results should be displayed
    expect(queryByText('Alice')).toBeNull();
    expect(queryByText('Bob')).toBeNull();
    expect(queryByText('Charlie')).toBeNull();
    expect(getByText('Data Not Found!')).toBeTruthy();
  });

  it('works correctly with filtered tabs', () => {
    const { getByPlaceholderText, getByTestId, queryByText } = render(
      <DashBoardScreen navigation={mockNavigation as any} route={mockRoute} />
    );

    const searchInput = getByPlaceholderText('Search by name...');
    const adminTab = getByTestId('tab-Admin');

    // Switch to Admin tab
    fireEvent.press(adminTab);

    // Type "Alice" in search
    fireEvent.changeText(searchInput, 'Alice');

    // Alice is visible, Charlie also Admin, Bob not visible
    expect(queryByText('Alice')).toBeTruthy();
    expect(queryByText('Charlie')).toBeTruthy();
    expect(queryByText('Bob')).toBeNull();
  });
});
