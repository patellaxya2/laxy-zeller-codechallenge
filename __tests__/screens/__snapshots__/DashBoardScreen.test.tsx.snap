// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should Snapshot match 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#FFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "margin": 10,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#F0F2F5",
          "borderRadius": 20,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "position": "relative",
          "width": "80%",
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "borderColor": "#007aff",
            "borderRadius": 20,
            "borderWidth": 1,
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "transform": [
              {
                "translateX": 3,
              },
            ],
            "width": 90,
            "zIndex": 1,
          }
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="tab-All"
      >
        <Text
          style={
            [
              {
                "color": "#333",
                "fontSize": 14,
              },
              {
                "color": "#007aff",
                "fontWeight": "600",
              },
            ]
          }
        >
          All
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="tab-Admin"
      >
        <Text
          style={
            [
              {
                "color": "#333",
                "fontSize": 14,
              },
              false,
            ]
          }
        >
          Admin
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="tab-Manager"
      >
        <Text
          style={
            [
              {
                "color": "#333",
                "fontSize": 14,
              },
              false,
            ]
          }
        >
          Manager
        </Text>
      </View>
    </View>
  </View>
  <TextInput
    onChangeText={[Function]}
    placeholder="Search by name..."
    style={
      {
        "alignSelf": "center",
        "borderColor": "#ccc",
        "borderRadius": 40,
        "borderWidth": 1,
        "marginVertical": 15,
        "padding": 10,
        "width": "85%",
      }
    }
  />
  <RNCViewPager
    initialPage={0}
    layoutDirection="ltr"
    onMoveShouldSetResponderCapture={[Function]}
    onPageScroll={[Function]}
    onPageScrollStateChanged={[Function]}
    onPageSelected={[Function]}
    style={
      {
        "backgroundColor": "#FFF",
        "flex": 1,
      }
    }
    testID="pager-view"
  >
    <View
      collapsable={false}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          [
            undefined,
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <RCTScrollView
          ListEmptyComponent={[Function]}
          contentContainerStyle={
            {
              "paddingBottom": 80,
            }
          }
          data={
            [
              {
                "data": [
                  {
                    "email": "alice@test.com",
                    "id": "1",
                    "name": "Alice",
                    "role": "ADMIN",
                  },
                ],
                "title": "A",
              },
              {
                "data": [
                  {
                    "email": "bob@test.com",
                    "id": "2",
                    "name": "Bob",
                    "role": "MANAGER",
                  },
                ],
                "title": "B",
              },
              {
                "data": [
                  {
                    "email": "charlie@test.com",
                    "id": "3",
                    "name": "Charlie",
                    "role": "ADMIN",
                  },
                ],
                "title": "C",
              },
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          renderItem={[Function]}
          scrollEventThrottle={0.0001}
          stickyHeaderIndices={
            [
              0,
              3,
              6,
            ]
          }
          testID="SectionList"
        >
          <View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <Text
                style={
                  {
                    "backgroundColor": "#f9f9f9",
                    "fontWeight": "bold",
                    "paddingHorizontal": 16,
                    "paddingVertical": 6,
                  }
                }
              >
                A
              </Text>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#ddd",
                    "flexDirection": "row",
                    "padding": 12,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#e6f0ff",
                      "borderRadius": 6,
                      "height": 32,
                      "justifyContent": "center",
                      "marginRight": 12,
                      "width": 32,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#007aff",
                        "fontWeight": "600",
                      }
                    }
                  >
                    A
                  </Text>
                </View>
                <Text
                  style={
                    {
                      "flex": 1,
                      "fontSize": 16,
                    }
                  }
                >
                  Alice
                </Text>
                <Text
                  style={
                    {
                      "color": "#999",
                      "fontSize": 12,
                    }
                  }
                >
                  Admin
                </Text>
              </View>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            />
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <Text
                style={
                  {
                    "backgroundColor": "#f9f9f9",
                    "fontWeight": "bold",
                    "paddingHorizontal": 16,
                    "paddingVertical": 6,
                  }
                }
              >
                B
              </Text>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#ddd",
                    "flexDirection": "row",
                    "padding": 12,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#e6f0ff",
                      "borderRadius": 6,
                      "height": 32,
                      "justifyContent": "center",
                      "marginRight": 12,
                      "width": 32,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#007aff",
                        "fontWeight": "600",
                      }
                    }
                  >
                    B
                  </Text>
                </View>
                <Text
                  style={
                    {
                      "flex": 1,
                      "fontSize": 16,
                    }
                  }
                >
                  Bob
                </Text>
              </View>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            />
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <Text
                style={
                  {
                    "backgroundColor": "#f9f9f9",
                    "fontWeight": "bold",
                    "paddingHorizontal": 16,
                    "paddingVertical": 6,
                  }
                }
              >
                C
              </Text>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#ddd",
                    "flexDirection": "row",
                    "padding": 12,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#e6f0ff",
                      "borderRadius": 6,
                      "height": 32,
                      "justifyContent": "center",
                      "marginRight": 12,
                      "width": 32,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#007aff",
                        "fontWeight": "600",
                      }
                    }
                  >
                    C
                  </Text>
                </View>
                <Text
                  style={
                    {
                      "flex": 1,
                      "fontSize": 16,
                    }
                  }
                >
                  Charlie
                </Text>
                <Text
                  style={
                    {
                      "color": "#999",
                      "fontSize": 12,
                    }
                  }
                >
                  Admin
                </Text>
              </View>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            />
          </View>
        </RCTScrollView>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          [
            undefined,
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      />
    </View>
    <View
      collapsable={false}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          [
            undefined,
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      />
    </View>
  </RNCViewPager>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#007aff",
        "borderRadius": 26,
        "bottom": 24,
        "elevation": 3,
        "height": 52,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 24,
        "width": 52,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 28,
          "textAlign": "center",
        }
      }
    >
      +
    </Text>
  </View>
</RNCSafeAreaView>
`;
